### DISCUSSION TITLE
TOC is not defined error on OpenAPI Docs

### DISCUSSION THREAD

**Discussion by olabrahmi** [fuma-nama/fumadocs#696] [Q&A] [answered]
I've been getting this error on OpenAPI docs pages after upgrading to `fumadocs-openapi@5.0.2`

<img width="987" alt="Screenshot 2024-08-12 at 13 20 12" src="https://github.com/user-attachments/assets/09192b19-a97b-44aa-af54-d5f6ec2e1606">

What am I doing wrong?

**Comment by fuma-nama** [comment]
Did you updated to the latest versions of Fumadocs? Make sure to re-generate your MDX files after upgrading Fumadocs OpenAPI

**Reply by olabrahmi** [reply]
Yeah actually, I've updated all fumadocs packages to the latest versions and re-generated the MDX files. Issue still occurring :(

**Reply by fuma-nama** [reply]
Oh you're right, the latest version doesn't compatible with Content Collections and other content sources because it generates table of contents and structured data on the server component. I'm making a fix which should fix the error.

**Reply by olabrahmi** [reply]
Thank you. I am using Remote MDX, I guess that's why it's not working for me. I'll wait for your fix

**Reply by fuma-nama** [reply, marked as answer]
Fumadocs OpenAPI 5.0.3 should work

**Reply by fuma-nama** [reply]
Just as a little reminder: I'm yet finding a way to expose the properties (e.g. `toc`) generated by Fumadocs OpenAPI, so that your content sources can access the output `toc` and `structuredData` even if remark plugins cannot find them.

Meanwhile `toc` and `structuredData` will probably be a Fumadocs MDX specific features, until I've time to implement that.
